src/Medidata.ZipkinTracer.Core/ZipkinClient.cs

```diff
@@ -34,7 +34,7 @@ static SpanCollector GetInstance(Uri uri, uint maxProcessorBatchSize)
             return instance;
         }
 
-        public ZipkinClient(IZipkinConfig zipkinConfig, IOwinContext context, SpanCollector collector = null)
+        public ZipkinClient(IZipkinConfig zipkinConfig, HttpContextBase context, SpanCollector collector = null)
```

Comment:
Yes, I adhoced with simultaneous incoming/outgoing requests, and with Owin and WebAPI as well.

Now one thing which I should note, that when you use the `ZipkinMessageHandler` for the ougoing requests, you have to provide the `HttpContext` yourself (more specifically `HttpContextBase`) because the handler won't do it for you - this is really easy in Owin and WebAPI, you just need to wrap the `HttpContext.Current` object: `new HttpContextWrapper(HttpContext.Current)` - as for command line applications, this is not trivial but it is not impossible either to have a working HttpContext (it is just very inconvenient).